app/src/main/java/io/gnosis/kouban/ui/transaction/details/TransactionDetailsViewModel.kt

```diff
@@ -45,30 +43,51 @@ class TransactionDetailsViewModel(
         }
 
     private fun Pair<Solidity.Address?, ServiceSafeTx>.toDetails(): List<Any> {
-        return listOf(
-            first!!,
-            second.tx.buildTransactionTypeView(first),
-            second.tx.to,
-            LabelDescription(
-                R.string.transaction_details_network_fees_label,
-                SpannableString(second.execInfo.fees.shiftedString(ETH_TOKEN_INFO.decimals, decimalsToDisplay = ETH_TOKEN_INFO.decimals))
-            ),
-            LabelDate(R.string.transaction_details_timestamp_label, dateInSecs = transaction.timestamp),
-            Link(transaction.executionHash!!, R.string.view_transaction_on),
-            LabelDescription(R.string.transaction_details_raw_data_label, SpannableString(second.tx.data))
-        )
+        val decodedData = decode(second.tx)
+        return mutableListOf<Any>().apply {
+            add(first!!)
+            add(second.buildTransactionTypeView(first, decodedData))
+            add(second.tx.to)
+            add(
+                LabelDescription(
+                    R.string.transaction_details_network_fees_label,
+                    SpannableString(second.execInfo.fees.shiftedString(ETH_TOKEN_INFO.decimals, decimalsToDisplay = ETH_TOKEN_INFO.decimals))
+                )
+            )
+            add(LabelDescription(R.string.transaction_details_operation_label, SpannableString(second.tx.operation.name)))
+            add(LabelDescription(R.string.transaction_details_value_label, SpannableString(second.tx.value.asDecimalString())))
+            second.txHash?.let { add(Link(it, R.string.view_transaction_on)) }
+            add(LabelDescription(R.string.transaction_details_raw_data_label, SpannableString(second.tx.data)))
+        }
     }
 
-    private fun SafeTx.buildTransactionTypeView(currentSafe: Solidity.Address?): TransactionTypeView =
+    private fun ServiceSafeTx.buildTransactionTypeView(currentSafe: Solidity.Address?, decodedData: DecodedData): TransactionTypeView =
```

Comment:
does distinction btw ServiceSafeTx and SafeTx makes sence? Should we combine them?

Reply:
I don't know I am just using the data layout that was already in place. I guess that the ServiceSafeTx from from the transaction API and SafeTx is just a mapping.